### PR TITLE
Icepak method

### PR DESCRIPTION
**Pull request by ansys-satyajeet**
Added two new methods to mechanical.py
* assign_heat_flux(objects_list, heat_flux_type, value, boundary_name="")
* assign_heat_generation(objects_list, value, boundary_name="")

Added create and update boundary lines into create and update methods of BoundaryObject class in Boundary.py

### CODE REVIEW

**Comment by ansys-satyajeet** [comment]
@maxcapodi78 Could you please check and let me know what is required from my end to resolve the code conflicts?

File: pyaedt/icepak.py [hunk 1]
```diff
@@ -2719,6 +2719,92 @@ def assign_surface_material(self, obj, mat):
                 self.materials.surface_material_keys[mat.lower()] = sm
         return True
 
+    @pyaedt_function_handler()
+    def create_two_resistor_network_block_depr(self, object_name, power, rjb, rjc, placement):
+        """Function to create 2-Resistor network object.
```
Comment (PipKat) [inline pyaedt/icepak.py:2724]:
```suggestion
        """Create a two-resistor network block.
```

File: pyaedt/icepak.py [hunk 2]
```diff
@@ -2719,6 +2719,92 @@ def assign_surface_material(self, obj, mat):
...
+
+        .. deprecated:: 0.6.30
+            This method will be replaced by `create_two_resistor_network_block`.
```
Comment (PipKat) [inline pyaedt/icepak.py:2727]:
```suggestion
            This method is replaced by the ``create_two_resistor_network_block`` method.
```

File: pyaedt/icepak.py [hunk 3]
```diff
@@ -2719,6 +2719,92 @@ def assign_surface_material(self, obj, mat):
...
+
+        Parameters
+        ----------
+        object_name : str
+            name of the object (3D block primitive) on which 2-R network is to be created
```
Comment (PipKat) [inline pyaedt/icepak.py:2732]:
```suggestion
            Name of the object (3D block primitive) on which to create the two-resistor network.
```

File: pyaedt/icepak.py [hunk 4]
```diff
@@ -2719,6 +2719,92 @@ def assign_surface_material(self, obj, mat):
...
+        power : float
+            junction power in [W]
+        rjb : float
+            Junction to board thermal resistance in [K/W]
+        rjc : float
+            Junction to case thermal resistance in [K/W]
+        placement : str
+            Acceptable entries are "top" or "bottom"
+            "top" : network block is placed on top of board
+            "bottom" : network block is placed on bottom of board
```
Comment (PipKat) [inline pyaedt/icepak.py:2742]:
```suggestion
        power : float
            Junction power in [W].
        rjb : float
            Junction-to-board thermal resistance in [K/W].
        rjc : float
            Junction-to-case thermal resistance in [K/W].
        placement : str
            Placement of the network block. Options are:
            - ``top``: Network block is placed on top of the board.
            - "bottom" : Network block is placed on bottom of the board.
```
Can we just say Options are ``top`` and ``bottom``? The descriptions seem unnecessary.

File: pyaedt/icepak.py [hunk 5]
```diff
@@ -2719,6 +2719,92 @@ def assign_surface_material(self, obj, mat):
...
+                    "Rjb": ["Board_side(" + board_side + ")", "Internal", "R", str(rjb) + "cel_per_w"],
+                }
+            ),
+            "SchematicData": ({}),
+        }
+
+        # Default material is Ceramic Material
+        self.modeler[object_name].material_name = "Ceramic_material"
+
+        # Create boundary condition and set Solve Inside to No
```
Comment (PipKat) [inline pyaedt/icepak.py:2800]:
```suggestion
        # Create boundary condition and set solve_inside = False
```

File: pyaedt/icepak.py [hunk 6]
```diff
@@ -2719,6 +2719,92 @@ def assign_surface_material(self, obj, mat):
...
+            "Links": OrderedDict(
+                {
+                    "Rjc": ["Case_side(" + case_side + ")", "Internal", "R", str(rjc) + "cel_per_w"],
+                    "Rjb": ["Board_side(" + board_side + ")", "Internal", "R", str(rjb) + "cel_per_w"],
+                }
+            ),
+            "SchematicData": ({}),
+        }
+
+        # Default material is Ceramic Material
```
Comment (PipKat) [inline pyaedt/icepak.py:2797]:
```suggestion
        # Default material is Ceramic_material
```

File: pyaedt/mechanical.py [hunk 7]
```diff
@@ -523,3 +523,95 @@ def existing_analysis_sweeps(self):
...
+    @pyaedt_function_handler()
+    def assign_heat_flux(self, objects_list, heat_flux_type, value, boundary_name=""):
+        """Assign heat flux boundary condition to an object or face list.
+
+        Parameters
+        ----------
+        objects_list : list
+            List of objects, faces, or both.
+        heat_flux_type : str
+            ``"Total Power"`` or ``"Surface Flux"``
```
Comment (PipKat) [inline pyaedt/mechanical.py:536]:
```suggestion
            Type of the heat flux. Options are ``"Total Power"`` or ``"Surface Flux"``.
```

File: pyaedt/mechanical.py [hunk 8]
```diff
@@ -523,3 +523,95 @@ def existing_analysis_sweeps(self):
...
+        value : str
+            Value of heat flux with units
```
Comment (PipKat) [inline pyaedt/mechanical.py:538]:
```suggestion
            Value of heat flux with units.
```

File: pyaedt/mechanical.py [hunk 9]
```diff
@@ -523,3 +523,95 @@ def existing_analysis_sweeps(self):
...
+
+        bound = BoundaryObject(self, boundary_name, props, "HeatFlux")
+        if bound.create():
+            self.boundaries.append(bound)
+            return bound
+        return False
+
+    @pyaedt_function_handler()
+    def assign_heat_generation(self, objects_list, value, boundary_name=""):
+        """Assign heat generation boundary condition to an object list.
```
Comment (PipKat) [inline pyaedt/mechanical.py:579]:
```suggestion
        """Assign a heat generation boundary condition to an object list.
```

**Review by maxcapodi78 — APPROVED** [review]
LGTM

**Pull request merged**